qiskit/transpiler/_transpiler.py

```diff
@@ -187,8 +187,10 @@ def transpile_dag(dag, basis_gates='u1,u2,u3,cx,id', coupling_map=None,
             dag = direction_mapper(dag, coupling)
             # Simplify cx gates
             cx_cancellation(dag)
+            # Unroll to the basis
+            dag = Unroller().run(dag)
```

Comment:
Why do we need this here? The Optimize1qGates() pass already does unrolling due to its `requires` attribute.
Also this line seems wrong since no basis is given, and it will thus unroll all the way to U, CX, which is not what we want.

Reply:
requires is only run when a passmanager is involved (it is the passmanager indeed who runs them). This does not happen when the pass is just run.